### ISSUE TITLE
When focused Intellisense item updates VoiceOver ignores that

### ISSUE THREAD

**Issue by keford** [microsoft/vscode#145550]
Issue Type: <b>Bug</b>

The incorrect information is now being read with VoiceOver when trying to use Intellisense.  Often you are hearing for example the closing tag of previously entered HTML.

1. Open a new file and set the type to HTML.
2. Enter that basic <html> tag.
3. Enter <he and use Intellisense to complete by down arrowing to the enter for <head>
4. On a new line enter <t.  Notice that what you hear is </head>

VS Code version: Code - Insiders 1.66.0-insider (Universal) (eaa2b431ac259fddf5aa11aa53060a9b70068b29, 2022-03-21T07:21:37.996Z)
OS version: Darwin arm64 21.4.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 3|
|Memory (System)|16.00GB (0.47GB free)|
|Process Argv|--crash-reporter-id 37b4596b-f9a0-4498-9026-c147121fb955|
|Screen Reader|yes|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
pythonvspyt551cf:30291413
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30404738
wslgetstarted:30449409
vsclayoutctrc:30405799
dsvsc009:30440021
pythonvsnew555:30438690
vscscmwlcmt:30428973
vscgsvid1:30433757
dockersurvey3cf:30438169
pynewfile477:30451556

```

</details>

**Comment by isidorn** [comment]
@keford thanks for filling this. However on step 4 it seems like you missed the end of your description.

Following your steps I end up in a file like this:
```
<header>
	<t
</header>
```

My cursor is on `<t` and as soon as I pressed `t` VoiceOver would read "header". Is that the bug you are trying to describe here?

**Comment by keford** [comment]
Apologies if this isn’t clear.  I will make a screen recording to illustrate what is happening.

Kelly


From: Isidor Nikolic ***@***.***>
Date: Monday, March 21, 2022 at 8:06 AM
To: microsoft/vscode ***@***.***>
Cc: Kelly Ford (CELA) ***@***.***>, Mention ***@***.***>
Subject: Re: [microsoft/vscode] Incorrect information being read with Intellisense and VoiceOver (Issue #145550)

@keford<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fkeford&data=04%7C01%7Ckeford%40microsoft.com%7Cad8d2cd4980848e661a608da0b3b8e50%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637834647665408800%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=nNbeHNG5OgZ2uc3l4LXejhZRm5HfS00ELm5eK%2FouLCw%3D&reserved=0> thanks for filling this. However on step 4 it seems like you missed the end of your description.

Following your steps I end up in a file like this:

<header>

        <t

</header>

My cursor is on <t and as soon as I pressed t VoiceOver would read "header". Is that the bug you are trying to describe here?

**Comment by isidorn** [comment]
@keford provided me more information in an email, and there are three issues here:
1. html suggests closing tags too aggressively. For that I opened https://github.com/microsoft/vscode/issues/145883
2. On suggest accept VoiceOver reads the whole editor content. We have informed VoiceOver people for this and they promised to take a look
3. On typing when suggest widget item focus changes that is not read out.

Let's continue discussing issue 3 here.
This seems to be a VoiceOver issue since it works just fine with NVDA (windows) and Orca (linux). Let me send an email to Apple and let you know what they say.

**Comment by isidorn** [comment]
@joanmarie I seem to be able to reproduce this also on Linux (not using latest Orca though). Steps:
1. Turn on Orca, start VS Code 
2. Cmd + n for new file, F1 > set language mode > javascript 
3. Type c, completion menu appears. Orca correctly reads the selected element in the completion menu
4. Type o -> completion menu gets updated. Orca does not read the newly selected item in the completion menu 🐛 

Can you try on your side when time allows. And if you can repro - do you know what events might be missing from Chromium that could help here?

**Comment by joanmarie** [comment]
Taking a look at what Orca gets, when I press 'c', there is a focus event for the list item (name: 'caches, docs: var caches: CacheStorageAvailable only in secure contexts.'). That's why Orca presents the first one.

If I press Down, I get a focus event for the newly-selected list item (name: 'cancelAnimationFrame, docs: function cancelAnimationFrame(handle: number): void). Subsequent presses of Down and Up work in the same fashion.

When I press "o", I see a lit of objects getting destroyed followed by name-changed events (even though the name appears to be the same). This is followed by a caret-moved event for the document being edited. It claims to be focused, so Orca presents the text it finds there ("co") and updates its location. BUT there is no new focus event for the newly-selected list item.

Same thing happens when I press "n". No new focus event.

Based on the above, my guess is that you're using active-descendant changes and got hit by a change in Chromium, which I worked around for Linux, and which may have been subsequently fixed in general. Can you repro this with VSCode Web using a pretty recent version of Chrome?

**Comment by isidorn** [comment]
@joanmarie thanks! Yes we are using `active-descendant` here.
I can reproduce using Chrome 100.0.4896.127 (Official Build) (64-bit) on my Linux (with an older Orca version).
On my macOS I tried Chrome Canary 103 and I could also reproduce with VoiceOver.

**Comment by isidorn** [comment]
I believe this should work much better with the latest vscode version. Thus I will go ahead and close this issue.
If you still encounter problems do let us know and we can reopen

fyi @meganrogge
